Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -32,8 +38,12 @@ public object Deserialize(IIonReader reader)
                 FieldInfo field;
                 if (property != null)
                 {
+                    if (!property.CanWrite)
```

Comment:
So read-only properties are never set on deserialization? Even if they were serialized?

Reply:
The issue is that `property.SetValue()` will not work for a read-only property because it has no setter.
To get around this we instead deserialize the read-only property's backing field. So the logic for handling read-only properties strangely happens during field processing instead of property processing.